src/aiperf/timing/strategies/adaptive_scale.py

```
@@ -0,0 +1,724 @@
+
+    async def _assessment_loop(self) -> None:
+        try:
+            while (
+                self._controller_phase != "complete"
+                and self._stop_checker.can_send_any_turn()
+            ):
+                await asyncio.sleep(self._assessment_period)
+                await self._assess_window()
+        except asyncio.CancelledError:
+            raise
+        except Exception as exc:
```

Ruff (0.15.17)

[warning] Do not catch blind exception: `Exception`

(BLE001)